lib/ecto_cursor_based_stream/task_synchronous.ex

```diff
@@ -0,0 +1,18 @@
+defmodule EctoCursorBasedStream.TaskSynchronous do
```

Comment:
It's just a Task, without any specific synchronization. Wouldn't be clearer if we named it just a `EctoCursorBasedStream.Task`?

Reply:
It is synchronous as it runs function in the same process in `async/2`: `result = fun.()`. It only mimics `Task` interface.